src/aead/aead.rs

```diff
@@ -613,4 +633,46 @@ mod tests {
                                         &mut in_out, &ad).is_err());
         }
     }
+
+    fn test_aead_inner(aead_alg: &'static aead::Algorithm,
+                       key_bytes: &[u8],
+                       nonce: &[u8],
+                       plaintext: &Vec<u8>,
+                       ad: &[u8],
+                       ct: &[u8],
+                       error: &Option<String>,
+                       in_prefix: &[u8]) {
+        let max_overhead_len = aead_alg.max_overhead_len();
+        let mut s_in_out = plaintext.clone();
+        for _ in 0..max_overhead_len {
+            s_in_out.push(0);
+        }
+        let s_key = aead::SealingKey::new(aead_alg, key_bytes).unwrap();
+        let s_result = aead::seal_in_place(&s_key, nonce,
+                                           &mut s_in_out[..],
+                                           max_overhead_len, &ad);
```

Comment:
All of the steps above only need to be done once per test vector in `test_aead_slow`, not once per `in_prefix_len`, right? Refactoring the test to pull out the invariant steps should speed this up a lot.